src/message/__tests__/messageActions-test.js

```diff
@@ -0,0 +1,58 @@
+import mockStore from 'redux-mock-store';
+
+import { doNarrow } from '../messagesActions';
+import { streamNarrow, homeNarrow } from '../../utils/narrow';
+
+const narrow = streamNarrow('some stream');
+const streamNarrowStr = JSON.stringify(narrow);
+const homeNarrowStr = JSON.stringify(homeNarrow());
+
+describe('messageActions', () => {
+  describe('doNarrow', () => {
+    test('when no messages in new narrow, actions to fetch messages and switch narrow are dispatched', () => {
```

Comment:
Not completely sure I got the point here. I guess you are implying that I also need to check for caughtUp state here https://github.com/zulip/zulip-mobile/blob/82be70669378739360b21c891dcef4ab147933be/src/message/messagesActions.js#L122 ?